src/webui/webapplication.cpp

```diff
@@ -774,6 +777,30 @@ void WebApplication::action_command_bottomPrio()
     BitTorrent::Session::instance()->bottomTorrentsPriority(hashes);
 }
 
+void WebApplication::action_command_setLocation()
+{
+    CHECK_URI(0);
+    CHECK_PARAMETERS("hashes" << "location");
+
+    QStringList hashes = request().posts["hashes"].split("|");
+    QString newLocation = request().posts["location"].trimmed();
+
+    // check location exists
+    if (!QDir(newLocation).exists())
```

Comment:
you need to check if newLocation is empty, otherwise QDir will check the program's working dir.